src/aal/aal_loongarch.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+#include <cstddef>
+#if _LOONGARCH_SZPTR == 64
+#  define SNMALLOC_VA_BITS_64
+#elif _LOONGARCH_SZPTR == 32
+#  define SNMALLOC_VA_BITS_32
+#else 
+#  error "Unsupported pointer size"
+#endif
+namespace snmalloc
+{
+  /**
+   * ARM-specific architecture abstraction layer.
+   */
+  class AAL_loongarch
+  {
+  public:
+    /**
+     * Bitmap of AalFeature flags
+     */
+    static constexpr uint64_t aal_features =
+      IntegerPointers | NoCpuCycleCounters;
+
+    static constexpr enum AalName aal_name = LoongArch;
+
+    static constexpr size_t smallest_page_size = 0x10000;
```

Comment:
This means 64K, which is not true on LoongArch; according to kernel sources this should be 0x1000 (4K, theoretical minimum page size) or 0x4000 (16K, vast majority of systems) instead. (I'm not familiar with the codebase, at least yet, so I don't currently know which is appropriate.)

Reply:
The results were based on my tests on a very early stage qemu, where `madvise` complains about page alignment for `0x1000`. But that simulator was buggy and I believe I should retest all these configurations on a recent simulator.